lib/src/Uncertainty/StatTests/VisualTest.cxx

```diff
@@ -540,4 +612,114 @@ Graph VisualTest::DrawKendallPlot(const Sample & firstSample,
   return graph;
 }
 
+/* Draw dependence functions */
+Graph VisualTestDrawDependenceFunction(const Sample & data,
+                                       const String & linkFormula,
+                                       const String & legend,
+                                       const Bool survival = false)
+{
+  if (!data.getSize())
+    throw InvalidArgumentException(HERE) << "The sample must not be empty";
+  if (data.getDimension() != 2)
+    throw InvalidArgumentException(HERE) << "The sample must be of dimension 2";
+  const UnsignedInteger size = data.getSize();
+  const Sample ranked((data.rank() + 1.0) / size);
+  Graph graph("n/a", "u", legend, true, "bottom");
+  const UnsignedInteger pointNumber = ResourceMap::GetAsUnsignedInteger("Evaluation-DefaultPointNumber");
+  Sample valuesU(pointNumber, 1);
+  Sample valuesXu(pointNumber, 1);
+  Sample valuesXuLow(pointNumber, 1);
+  Sample valuesXuUp(pointNumber, 1);
+  const Scalar level = ResourceMap::GetAsScalar("VisualTest-DependenceConfidenceLevel");
+  const Scalar xq = DistFunc::qNormal(0.5 + 0.5 * level);
+  const SymbolicFunction link(Description({"u", "cuu"}), Description(1, linkFormula));
+  for (UnsignedInteger i = 0; i < pointNumber; ++ i)
+  {
+    const Scalar u = (i + 1) * 1.0 / (pointNumber + 1);
+    valuesU(i, 0) = u;
+    Scalar cuu = 0.0;
+    for (UnsignedInteger j = 0; j < size; ++ j)
+    {
+      if (survival)
+      {
+        if (u < std::min(ranked(j, 0), ranked(j, 1)))
+          cuu += 1.0 / size;
+      }
+      else
+      {
+        if (u > std::max(ranked(j, 0), ranked(j, 1)))
+          cuu += 1.0 / size;
+      }
+    }
+    if ((cuu > 0.0) && (cuu < 1.0))
+    {
+      // estimate
+      valuesXu[i] = link(Point({u, cuu}));
+
+      // confidence interval of cuu, then xu
+      Scalar lb = cuu - xq * std::sqrt((cuu * (1.0 - cuu)) / size);
+      // might be < 0, set something positive below cuu
+      const Scalar epsilon = 1e-3;
```

Comment:
A more explicit comment on the 1e-3 value?

Reply:
better use specfuncprecision